### ISSUE TITLE
Version 2.0.2 cannot open file

### ISSUE THREAD

**Issue by prubin73** [linuxmint/xreader#276]
```
 * Xreader version (xreader --version)
2.0.2
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Mint 19.1
```

**Issue**
Attempting to open the attached file, xreader 2.0.2 crashes with the error "double free or corruption (out)". On Mint 18.3, xreader 1.6.2 has no trouble opening the same file.

**Steps to reproduce**
Try to open the attached file.

**Expected behaviour**
File should open.
[test.pdf](https://github.com/linuxmint/xreader/files/2757223/test.pdf)

**Other information**

**Comment by icarter09** [comment]
Linux Mint 19.2 64bit
Xreader v2.2.3

Was able to reproduce this issue.

**Comment by okaestne** [comment]
Can't reproduce on LM20 (Xreader v2.6.4 as well as v2.2.3) using the given file. So I guess it's some dependency's fault?

**Comment by prubin73** [comment]
It gets weirder. I just tried Xreader 2.4.4 on Mint 19.3. If I open Xreader in a terminal and then open the PDF file via File > Open, the contents display. (There are a bunch of failed assertions in the terminal, but they are apparently harmless.) On the other hand, if I double click the file in Caja it opens about one time in six, with Xreader failing the other times. If I open it with "xreader test.pdf" in a terminal, I get the same set of assertions followed by

```
double free or corruption (out)
Aborted (core dumped)
```
and the Xreader window disappears immediately after appearing.

So as of Xreader 2.4.4/Mint 19.3, it still crashes (except when it doesn't). That said, I also tried Xreader 2.6.4 on Mint 20 and, as @okaestne said, it works fine there. (In fact, the assertions are gone as well.)
